src/jllvm/object/StringInterner.hpp

```diff
@@ -28,21 +24,20 @@ class StringInterner
 
     llvm::DenseMap<std::pair<llvm::ArrayRef<std::uint8_t>, std::uint8_t>, String*> m_contentToStringMap;
     llvm::BumpPtrAllocator m_allocator;
-    ClassLoader& m_classLoader;
-    ClassObject* m_stringClass{nullptr};
+    ClassObject* m_byteArrayClass{};
+    ClassObject* m_stringClass{};
 
     void checkStructure();
 
     String* createString(llvm::ArrayRef<std::uint8_t> buffer, jllvm::CompactEncoding encoding);
 
 public:
-    StringInterner(ClassLoader& classLoader) : m_classLoader(classLoader) {}
-
-    void loadStringClass();
-
-    ClassObject& getStringClass() const
+    template <std::invocable<FieldType> F>
+    void initialize(F&& initializer)
```

Comment:
Coud you leave a doc comment here describing what this does, when it has to be called (prior to ever using `intern`), and what the requirements of `initializer` are (i.e. that it should return a class object of the given field type)